### ISSUE TITLE
Bottom sliver of playback bar left behind when scrolling

### ISSUE THREAD

**Issue by basaigh** [musescore/MuseScore#22998]
### Issue type

UI bug (incorrect info or interface appearance)

### Bug description

The zoom level for which this occurs seems to be inconsistent, even between systems of the same score.
For replicability, on the attached score a zoom of 449% should provide a good example for the first system at least.

### Steps to reproduce

1. Open a score.
2. Press play and start zooming in and out.
3. See that for some zoom levels the very bottom of the playback bar is left behind.
4. Observe that scrolling vertically or horizontally causes the position of the sliver to update.

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/53559772/5662007e-f111-4377-81e1-84f89316df41

### MuseScore Version

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.3.1-241421429, revision: github-musescore-musescore-2544ca1

### Regression

I don't know

### Operating system

Windows 11

### Additional context

This also appears to occur on MuseScore Studio 4.3.0, I don't know, however, when it began.

This also feels like it should be a duplicate, but I couldn't find a pre-existing issue, sorry if I missed it.

[Mendelssohn Sonata.zip](https://github.com/musescore/MuseScore/files/15461874/Mendelssohn.Sonata.zip)

**Comment by wizofaus** [comment]
Yes I've observed this, and while I have fixed some related clipping issues, my initial fix only made things worse, and I was happy enough for my final fix to at least not do that, but this does still occur, particularly if you zoom in & out while playing back.  Unfortunately it's quite hard to figure out why this might be happening, I'm not sure it's worth the investment in time, when it may well be related to combination of bugs in Qt and/or OS specific behaviours, and doesn't really cause any significant inconvenience.

**Comment by bkunda** [comment]
FWIW I can only reproduce this on Windows.
